presto-main/src/test/java/com/facebook/presto/execution/resourceGroups/TestResourceGroups.java

```diff
@@ -148,4 +157,67 @@ public void testMemoryLimit()
         assertEquals(query2.getState(), RUNNING);
         assertEquals(query3.getState(), RUNNING);
     }
+
+    @Test(timeOut = 10_000)
+    public void testWeightedScheduling()
+    {
+        RootResourceGroup root = new RootResourceGroup("root", directExecutor());
+        root.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        root.setMaxQueuedQueries(4);
+        // Start with zero capacity, so that nothing starts running until we've added all the queries
+        root.setMaxRunningQueries(0);
+        root.setSchedulingPolicy(WEIGHTED);
+        ResourceGroup group1 = root.getOrCreateSubGroup("1");
+        group1.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        group1.setMaxQueuedQueries(2);
+        group1.setMaxRunningQueries(2);
+        ResourceGroup group2 = root.getOrCreateSubGroup("2");
+        group2.setSoftMemoryLimit(new DataSize(1, MEGABYTE));
+        group2.setMaxQueuedQueries(2);
+        group2.setMaxRunningQueries(2);
+        group2.setSchedulingWeight(2);
+
+        Set<MockQueryExecution> group1Queries = fillGroup(group1, ImmutableSet.of());
+        Set<MockQueryExecution> group2Queries = fillGroup(group2, ImmutableSet.of());
+        root.setMaxRunningQueries(1);
+
+        int group2Ran = 0;
+        for (int i = 0; i < 1000; i++) {
+            for (Iterator<MockQueryExecution> iterator = group1Queries.iterator(); iterator.hasNext(); ) {
+                MockQueryExecution query = iterator.next();
+                if (query.getState() == RUNNING) {
+                    query.complete();
+                    iterator.remove();
+                }
+            }
+            for (Iterator<MockQueryExecution> iterator = group2Queries.iterator(); iterator.hasNext(); ) {
+                MockQueryExecution query = iterator.next();
+                if (query.getState() == RUNNING) {
+                    query.complete();
+                    iterator.remove();
+                    group2Ran++;
+                }
+            }
+            root.processQueuedQueries();
+            group1Queries = fillGroup(group1, group1Queries);
+            group2Queries = fillGroup(group2, group2Queries);
+        }
+
+        BinomialDistribution binomial = new BinomialDistribution(1000, 2.0 / 3.0);
```

Comment:
maybe add a comment so the next reader doesn't have to decipher the code... something like
```
// group1 has a weight of 1 and group2 has a weight of 2 so group2 should (2 / (1 + 2)) of the queries.
// since this is stochastic, we are check the result of 1000 trials are 2/3 with 99.9999 confidence